sdk/identity/perf-tests/identity/test/ClientSecretCredential/persistence.spec.ts

```diff
@@ -0,0 +1,34 @@
+import { PerfStressTest } from "@azure/test-utils-perfstress";
+import { ClientSecretCredential } from "@azure/identity";
+
+const scope = `https://servicebus.azure.net/.default`;
+
+/**
+ * This test does silent authentication with persistence enabled.
+ */
+export class ClientSecretCredentialPersistenceTest extends PerfStressTest {
+  options = {};
+  credential: ClientSecretCredential;
+
+  constructor() {
+    super();
+    const tenantId = process.env.AZURE_TENANT_ID!;
+    const clientId = process.env.AZURE_CLIENT_ID!;
+    const clientSecret = process.env.AZURE_CLIENT_SECRET!;
+
+    this.credential = new ClientSecretCredential(tenantId, clientId, clientSecret, {
```

Comment:
**Just some information..**
Perf framework provides a "parallel" option that instantiates this test class ClientSecretCredentialPersistenceTest "parallel" number of times.
So, with the current test, each instantiation will create a new credential per parallel instance. (If you want to test the same credential across all the parallels, you can declare it outside or make it a static variable.)

Coming to the globalSetup, it is run only for one of the instantiated classes.. meaning the resources created at the globalSetup are meant to be shared across the parallel instances.

Reply:
@sadasant Have you tried the parallel flag?
I'm unsure if the test works correctly based on what I said above.

Reply:
gotcha, I'll move the credential to global setup! thanks!